cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -1715,49 +1343,38 @@ Slice::JavaGenerator::writeSequenceMarshalUnmarshalCode(
     string typeS = typeToString(seq, TypeModeIn, package);
     string v = param;
 
-    //
-    // If the sequence is a byte sequence, check if there's the serializable metadata to
-    // get rid of this case first.
-    //
-    BuiltinPtr builtin = dynamic_pointer_cast<Builtin>(seq->type());
-    if (builtin && builtin->kind() == Builtin::KindByte)
```

Comment:
The validator already ensures that `java:serializable`  only goes on byte sequences.
No need to repeat the validation here.